src/core/ready-no-deferred.js

```diff
@@ -0,0 +1,109 @@
+define( [
+	"../core",
+	"../var/document"
+], function( jQuery, document ) {
+
+var readyCallbacks = [],
+	readyFiring = false,
+	whenReady = function( fn ) {
+		readyCallbacks.push( fn );
+	},
+	executeReady = function( fn ) {
+
+		// Prevent errors from freezing future callback execution (gh-1823)
+		// Not backwards-compatible as this does not execute sync
+		window.setTimeout( function() {
+			fn.call( document, jQuery );
```

Comment:
Let's open another ticket then. This PR is already 2 tickets. I'd rather not hold it up. But I'm indifferent as to what context we use here. `document` still makes sense since it's the document ready handler.